### PR TITLE
API-5015: Audit Special Issues/Flashes Enum values for EVSS and BGS

### PR DESCRIPTION
**Pull request by SlakrHakr**
## Description of change
Maps special issues sent to us to values EVSS expects. This change also removes any special issues EVSS does not accept from our payload to EVSS. (Those additional values will be added through BGS in an existing subsequent background job)

## Original issue(s)
https://vajira.max.gov/browse/API-5015

## Things to know about this PR
Tested locally and added specs

### CODE REVIEW

File: modules/claims_api/lib/claims_api/special_issue_mappers/bgs.rb [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+          { name: 'VASRD-Mental', code: 'VML' },
+          { name: 'VASRD-Musculoskeletal', code: 'VM' },
+          { name: 'VASRD-Neurological', code: 'VN' },
+          { name: 'VASRD-Respiratory/Auditory', code: 'VRA' },
+          { name: 'VASRD-Skin', code: 'VS' },
+          { name: 'Vendor Exclusion - No Diagnosis', code: 'VEXCLNODIAG' },
+          { name: 'VONAPP Direct Connect', code: 'VDC' },
+          { name: 'WARTAC', code: 'WARTAC' },
+          { name: 'WARTAC Trainee', code: 'WT' }
+        ] + super).uniq { |special_issue| special_issue[:name] }
```
Comment (jw81) [inline modules/claims_api/lib/claims_api/special_issue_mappers/bgs.rb:163]: Should we freeze this collection like we do in the parent?

File: modules/claims_api/spec/lib/claims_api/special_issue_mappers/bgs_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  [
+    { name: 'Character of Discharge', code: 'CD' },
+    { name: 'PTSD/1', code: 'PTSD/1' }
+  ].each do |special_issue|
+    it "returns correct code for name: #{special_issue[:name]}" do
+      expect(subject.code_from_name!(special_issue[:name])).to eq(special_issue[:code])
+    end
+  end
+
+  it 'returns nil for invalid name' do
```
Comment (jw81) [inline modules/claims_api/spec/lib/claims_api/special_issue_mappers/bgs_spec.rb:16]: Could have a test for the ! version to make sure it raises.  (but ya don't have to 😉)

Comment (SlakrHakr) [reply modules/claims_api/spec/lib/claims_api/special_issue_mappers/bgs_spec.rb:16]: https://github.com/department-of-veterans-affairs/vets-api/pull/6013/files#diff-f667abb030578efac5a8fcc97cb99d00d7fb418afda7a04065f97db46470b291R20

and

https://github.com/department-of-veterans-affairs/vets-api/pull/6014/files#diff-b4fd875aa5bb80c833d125d38929e9fb6528fc260574645381dd4fd811d59d30R20

File: modules/claims_api/lib/claims_api/special_issue_mappers/bgs.rb (hunk 1 above)
Comment (SlakrHakr) [reply modules/claims_api/lib/claims_api/special_issue_mappers/bgs.rb:163]: https://github.com/department-of-veterans-affairs/vets-api/pull/6014/files#diff-a09bad8abe02f0b3548627a2840903b06a8424e211cbd9980773b6108ed55269R163

**Comment by SlakrHakr** [comment]
Broke this out into 2 separate prs:
https://github.com/department-of-veterans-affairs/vets-api/pull/6013
and
https://github.com/department-of-veterans-affairs/vets-api/pull/6014

**Pull request closed without merging**